File: star-pattern-2/python/StarPattern2.py
def main():
    number = int(input('Enter a number: '))

    if (number <= 0):
        print('Invalid input. Please enter a positive number greater than zero.')
    else:
        print('You have entered number:', number)
        for i in range(number, 0, -1):
            print(i * '* ')

if __name__ == '__main__':
    print('--- Star Pattern 2 ---')
    main()

File: star-pattern-2/java/StarPattern2.java
import java.util.Scanner;

public class StarPattern2 {
    public static void main(String[] args) {
        System.out.println("--- Star Pattern 2 ---");

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number: ");
        var input = scanner.nextInt();

        if (input <= 0)
            System.out.println("Invalid input. Please enter a positive number greater than zero.");
        else {
            System.out.println("You have entered number: " + input);
            for (int i = 1; i <= input; i++) {
                for (int j = input; j >= i; j--)
                    System.out.print("* ");
                System.out.println("\t");
            }
        }

        scanner.close();
    }
}